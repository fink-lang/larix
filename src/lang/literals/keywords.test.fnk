{describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'

{parse} = import '../../parser.fnk'


describe 'keywords', fn:
  it 'parses true', fn:
    expect
      parse '
        foo = true
        bar = trueish
      '
      to_match_snapshot


  it 'parses false', fn:
    expect
      parse '
        foo = false
        bar = falseish
      '
      to_match_snapshot


  it 'parses inside exprs', fn:
    expect
      parse '
        [false, true]
        {false: true, true: false}
      '
      to_match_snapshot
