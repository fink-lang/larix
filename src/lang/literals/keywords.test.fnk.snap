// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keywords parses false 1`] = `
module (1:0-3:0)
  assign = (1:0-1:11)
    ident (1:0-1:3) foo
    literal (1:6-1:11) false
  assign = (2:0-2:14)
    ident (2:0-2:3) bar
    ident (2:6-2:14) falseish
`;

exports[`keywords parses null 1`] = `
module (1:0-3:0)
  assign = (1:0-1:10)
    ident (1:0-1:3) foo
    literal (1:6-1:10) null
  assign = (2:0-2:13)
    ident (2:0-2:3) bar
    ident (2:6-2:13) nullish
`;

exports[`keywords parses true 1`] = `
module (1:0-3:0)
  assign = (1:0-1:10)
    ident (1:0-1:3) foo
    literal (1:6-1:10) true
  assign = (2:0-2:13)
    ident (2:0-2:3) bar
    ident (2:6-2:13) trueish
`;
